src/agent/sqlite_manager/src/sqlite_manager.cpp

```diff
@@ -0,0 +1,289 @@
+        queryString.erase(queryString.size() - 1);
+        queryString += ") VALUES (";
+
+        for (const Col& col : cols)
+        {
+            if (col.m_type == SQLite::TEXT)
+                queryString += "\"";
+            queryString += col.m_value;
+            if (col.m_type == SQLite::TEXT)
+                queryString += "\"";
+            queryString += ",";
+        }
+        // Remove the trailing ", "
+        queryString.erase(queryString.size() - 1);
+        queryString += ");";
+
+        std::lock_guard<std::mutex> lock(m_mutex);
+        try
+        {
+            m_db->exec(queryString);
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Error inserting element: " << e.what() << std::endl;
+            throw;
+        }
+    }
+
+    int SQLiteManager::GetCount(const std::string& tableName)
+    {
+        std::string queryString = std::format("SELECT COUNT(*) FROM {}", tableName);
+
+        try
+        {
+            SQLite::Statement query(*m_db, queryString);
+            int count = 0;
+            if (query.executeStep())
+            {
+                count = query.getColumn(0).getInt();
+            }
+            else
+            {
+                std::cerr << "Error getting element count." << std::endl;
+            }
+            return count;
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Error during GetCount operation: " << e.what() << std::endl;
+            return {};
+        }
+    }
+
+    void SQLiteManager::ExecuteNoSelectSQL(const std::string& queryString)
+    {
+        std::lock_guard<std::mutex> lock(m_mutex);
+        try
+        {
+            m_db->exec(queryString);
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Error executing SQL: " << e.what() << std::endl;
+            throw;
+        }
+    }
+
+    std::vector<Row> SQLiteManager::Select(const std::string& tableName,
+                                           const std::vector<Col>& fields,
+                                           const std::vector<Col>& selCriteria)
+    {
+        // Build the query string
+        std::string queryString;
+        if (fields.size() == 0)
+        {
+            queryString = std::format("SELECT * FROM {}", tableName);
+        }
+        else
+        {
+            queryString = "SELECT ";
+            for (auto& col : fields)
+            {
+                queryString += col.m_name + ", ";
+            }
+
+            // Remove the trailing ", "
+            queryString.erase(queryString.size() - 2);
+            queryString += " FROM " + tableName;
+        }
+
+        if (selCriteria.size() != 0)
+        {
+            queryString += " WHERE ";
+            for (auto& col : selCriteria)
+            {
+                if (col.m_type == SQLite::TEXT)
+                {
+                    queryString += col.m_name + "='" + col.m_value + "' AND ";
```

Comment:
What about the `OR`?

Reply:
Added extra parameter that allows the caller to specify the operator AND or OR. Default is AND.